### PR TITLE
test_element: Add test cases for Config::Element

### PR DESCRIPTION
**Pull request by cosmo0920**
Added these test cases:

* `#initialize`
* `#add_element`
* `#==`
* `#+`
* `#has_key?`
* `#to_s`
* `#inspect`

### CODE REVIEW

File: test/config/test_element.rb [hunk 1]
```diff
@@ -63,4 +63,120 @@ def element(name = 'ROOT', arg = '', attrs = {}, elements = [])
...
+                       element('test', 'mydata', {'k3' => 'v3'}, [])
+                     ])
+        assert_false(e1 == e2)
+        assert_false(e2 == e3)
+        assert_false(e3 == e4)
+      end
+    end
+  end
+
+  sub_test_case '#+' do
```
Comment (tagomoris) [inline test/config/test_element.rb]: It's better to have tests like:
* overwriting existing attributes in e1 by e2
* adding elements in e2 to elements of e1

File: test/config/test_element.rb [hunk 2]
```diff
@@ -63,4 +63,120 @@ def element(name = 'ROOT', arg = '', attrs = {}, elements = [])
...
+  sub_test_case '#has_key?' do
+    test 'can get boolean with key name' do
+      e = element('ROOT', 'mydata', {"k1" => "v1"}, [])
+      assert_true(e.has_key?("k1"))
+      assert_false(e.has_key?("noexistent"))
+    end
+  end
+
+  sub_test_case '#to_s' do
+    test 'dump config element with #to_s' do
```
Comment (tagomoris) [inline test/config/test_element.rb]: Please add another test case for instance with `@v1_config = true`.

**Comment by tagomoris** [comment]
@cosmo0920 Do you have plan to add another test cases on this pull-request for cases below? Or is that all at this time?
* `check_not_fetched` and `unused`
* `to_masked_element`, `#secret_param?`, `param_type`

**Comment by cosmo0920** [comment]
I think that:

* `to_masked_element`, `#secret_param?`, `param_type`

are implicitly tested in `secret parameter` feature test case(s).

`check_not_fetched` and `unused` might be written test cases in this PR.

**Comment by tagomoris** [comment]
Great. Thank you!

**Pull request merged**